src/calendar-v1/CalendarCell.ts

```diff
@@ -36,23 +32,64 @@ export const useCalendarCell = createHook<
   compose: useBox,
   keys: CALENDAR_CELL_KEYS,
 
-  useProps({ date, getCellOptions, highlightDate, isDisabled }, htmlProps) {
-    const cellProps = getCellOptions?.(date);
+  useProps(options, { onMouseEnter: htmlOnMouseEnter, ...htmlProps }) {
+    const {
+      date,
+      dateValue,
+      highlightDate,
+      highlightedRange,
+      isDisabled,
+      currentMonth,
+    } = options;
 
-    const onMouseEnter = () => {
-      highlightDate && highlightDate(date);
+    let calendarProps: any = {
+      role: "gridcell",
+      "data-weekend": dataAttr(isWeekend(date)),
     };
 
+    if ("highlightDate" in options) {
+      const isSelected = highlightedRange
+        ? date >= highlightedRange.start && date <= highlightedRange.end
+        : false;
+
+      const isRangeStart = isSelected && date.getDate() === 1;
+      const isRangeEnd =
+        isSelected && date.getDate() === getDaysInMonth(currentMonth);
+      const isSelectionStart = highlightedRange
+        ? isSameDay(date, highlightedRange.start)
+        : false;
+      const isSelectionEnd = highlightedRange
+        ? isSameDay(date, highlightedRange.end)
+        : false;
+
+      const onMouseEnter = () => {
+        if (isDisabled) return;
+
+        highlightDate?.(date);
+      };
```

Comment:
I kinda liked that this chunk of logic was in the state return `getCellOptions`, it's noisy in here. 🤔 maybe we can refactor/extract is somewhere.

Reply:
But I thought it's not necessary to be in a state.
Because they are specific to the component.

May be we can refactor it.

Reply:
> But I thought it's not necessary to be in a state.
> Because they are specific to the component.
> 
> May be we can refactor it.

Yeah, although it was nicer since it was lazily in the state wasn't evaluated and has access to all the variables already since it had access to it's parent scope. just cleaner in general. we can maybe also extract it into another function if you don't want to put it on the state.

Reply:
Yup, your points are vaild, one adv is it had access to all the variables.
I don't want them to be in the state because they don't belong there.
I have refactored it a bit, help me if you had a better idea.

Reply:
Removing them, made the state cleaner